test/integration/scheduler/queue_test.go

```diff
@@ -507,6 +486,131 @@ func TestCoreResourceEnqueue(t *testing.T) {
+			},
+			pods: []*v1.Pod{
+				// - Pod3 and Pod4 will be rejected by the PodTopologySpread plugin.
+				st.MakePod().Name("pod3").Label("key1", "val").SpreadConstraint(1, "node", v1.DoNotSchedule, st.MakeLabelSelector().Exists("key1").Obj(), ptr.To(int32(2)), nil, nil, nil).Container("image").Obj(),
+				st.MakePod().Name("pod4").Label("key1", "val").SpreadConstraint(1, "zone", v1.DoNotSchedule, st.MakeLabelSelector().Exists("key1").Obj(), ptr.To(int32(2)), nil, nil, nil).Container("image").Obj(),
+			},
+			triggerFn: func(testCtx *testutils.TestContext) error {
+				// Trigger an assigned Node add event.
+				// It should requeue pod3 only because this node only has node label, and doesn't have zone label that pod4 requires.
+				node := st.MakeNode().Name("fake-node3").Label("node", "fake-node").Obj()
+				if _, err := testCtx.ClientSet.CoreV1().Nodes().Create(testCtx.Ctx, node, metav1.CreateOptions{}); err != nil {
+					return fmt.Errorf("failed to create a new node: %w", err)
+				}
+				return nil
+			},
+			wantRequeuedPods:          sets.New("pod3"),
+			enableSchedulingQueueHint: []bool{true},
+		},
+		{
+			name: "Pods with PodTopologySpread should be requeued when a Node label is updated",
+			initialNodes: []*v1.Node{
+				st.MakeNode().Name("fake-node1").Label("node", "fake-node").Obj(),
+				st.MakeNode().Name("fake-node2").Label("node", "fake-node").Obj(),
+			},
+			initialPods: []*v1.Pod{
+				st.MakePod().Name("pod1").Label("key1", "val").SpreadConstraint(1, "node", v1.DoNotSchedule, st.MakeLabelSelector().Exists("key1").Obj(), nil, nil, nil, nil).Container("image").Node("fake-node1").Obj(),
+				st.MakePod().Name("pod2").Label("key1", "val").SpreadConstraint(1, "zone", v1.DoNotSchedule, st.MakeLabelSelector().Exists("key1").Obj(), nil, nil, nil, nil).Container("image").Node("fake-node2").Obj(),
+			},
+			pods: []*v1.Pod{
+				// - Pod3 and Pod4 will be rejected by the PodTopologySpread plugin.
+				st.MakePod().Name("pod3").Label("key1", "val").SpreadConstraint(1, "node", v1.DoNotSchedule, st.MakeLabelSelector().Exists("key1").Obj(), ptr.To(int32(3)), nil, nil, nil).Container("image").Obj(),
+				st.MakePod().Name("pod4").Label("key1", "val").SpreadConstraint(1, "zone", v1.DoNotSchedule, st.MakeLabelSelector().Exists("key1").Obj(), ptr.To(int32(3)), nil, nil, nil).Container("image").Obj(),
+			},
+			triggerFn: func(testCtx *testutils.TestContext) error {
+				// Trigger an assigned Node update event.
+				// It should requeue pod4 only because this node only has zone label, and doesn't have node label that pod3 requires.
+				node := st.MakeNode().Name("fake-node2").Label("zone", "fake-node").Obj()
+				if _, err := testCtx.ClientSet.CoreV1().Nodes().Update(testCtx.Ctx, node, metav1.UpdateOptions{}); err != nil {
+					return fmt.Errorf("failed to update node: %w", err)
+				}
+				return nil
+			},
+			wantRequeuedPods:          sets.New("pod4"),
+			enableSchedulingQueueHint: []bool{true},
+		},
```

Comment:
Looks like we have two missing test cases for -
- Node/Delete: Pods with PodTopologySpread should be requeued when a deleted node had the topology label
- Node/UpdateNodeTaint: Pods with PodTopologySpread should be requeued when untainted Node has the topology label

Reply:
Added test cases related to Node/UpdateNodeTaint.

I am unable to create test cases for Node/Delete.
In the case of Node deletion, `isSchedulableAfterNodeChange` in `EventsToRegister` is not executed. I checked it locally, and that was the case.
I believe that for it to be executed, the MoveAllToActiveOrBackoffQueue function needs to be triggered in the `DeleteFunc: sched.deletePodFromSchedulingQueue` when a Node is deleted in `addAllEventHandlers.` However, unlike `AddFunc` and `UpdateFunc,` it is not being executed.
https://github.com/kubernetes/kubernetes/blob/bf3f6518610f7f5a39fc7842424a2c43350273f3/pkg/scheduler/eventhandlers.go#L419

If there is anything I've overlooked, please let me know.

Reply:
Ugh, that's the bug [we've found](https://github.com/kubernetes/kubernetes/issues/110175#issuecomment-1140397251) and forgot to fix..
Okay, for now, let's just skip adding Node/Delete test case. I'll ping you after I give a patch and you can implement Node/Delete test case then.